src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,164 @@
+                "Got it. I've added this task:\n%s\nNow you have %d tasks in the list.",
+                t, this.taskApplication.getNoOfTasks()
+        );
+    }
+
+    private String deadlineCommand(String title, String deadline) {
+        Task t;
+        t = new Deadline(title, deadline);
+        this.taskApplication.addTask(t);
+        return String.format(
+                "Got it. I've added this task:\n%s\nNow you have %d tasks in the list.",
+                t, this.taskApplication.getNoOfTasks()
+        );
+    }
+
+    private String eventCommand(String title, String start, String end) {
+        Task t = new Event(title, start, end);
+        this.taskApplication.addTask(t);
+        return String.format(
+                "Got it. I've added this task:\n%s\nNow you have %d tasks in the list.",
+                t, this.taskApplication.getNoOfTasks()
+        );
+    }
+
+    private String deleteCommand(int index) throws DukeException {
+        Task t = this.taskApplication.popTask(index);
+        return String.format(
+                "Noted. I've removed this task:\n%s\nNow you have %d tasks in the list.",
+                t, this.taskApplication.getNoOfTasks()
+        );
+    }
+    private String findCommand(String keyword) {
+        List<Task> tasks = this.taskApplication.getTaskByKeyword(keyword);
+        StringBuilder result = new StringBuilder();
+        int i = 1;
+        for (Task s: tasks) {
+            result.append(String.format("%d.%s\n", i, s));
+            i++;
+        }
+        String output = result.toString().replaceAll("\\n$", "");
+        return String.format("Here are the matching tasks in your list:\n%s", output);
+    }
+
+    private String byeCommand() {
+        this.taskApplication.close();
+        this.isExit = true;
+        return "Bye. Hope to see you again soon!";
+    }
+    private String parseCommand(String command) throws DukeException {
+        List<String> tokens = this.parser.parseCommand(command);
+        switch(tokens.get(0)) {
+            case "list":
+                return this.listTasksCommand();
+            case "mark":
+                return this.markCommand(Integer.parseInt(tokens.get(1)) - 1);
+            case "unmark":
+                return this.unmarkCommand(Integer.parseInt(tokens.get(1)) - 1);
+            case "todo":
+                return this.todoCommand(tokens.get(1));
+            case "deadline":
+                return this.deadlineCommand(tokens.get(1), tokens.get(2));
+            case "event":
+                return this.eventCommand(tokens.get(1), tokens.get(2), tokens.get(3));
+            case "delete":
+                return this.deleteCommand(Integer.parseInt(tokens.get(1)) - 1);
+            case "find":
+                return this.findCommand(tokens.get(1));
+            case "bye":
+                return this.byeCommand();
+            default:
+                throw new DukeUnknownCommandException("Unknown command");
```

Comment:
Your code looks good and it is very easy to follow! Perhaps you can consider declare the commands (list, unmark, mark, todo...) to constants to follow the coding standard!